### ISSUE TITLE
Experiment with Inspection reports instead of Test reports

### ISSUE THREAD

**Issue by andreogle** [andreogle/eslint-teamcity#11]
https://github.com/andreogle/eslint-teamcity/issues/10

**Comment by andreogle** [comment]
Reference from previous issue: https://github.com/andreogle/eslint-teamcity/issues/10

https://confluence.jetbrains.com/display/TCD10/Build+Script+Interaction+with+TeamCity#BuildScriptInteractionwithTeamCity-Inspectiontype
we may report eslint violations as inspection errors, not tests
If we are use that we must push any new error type as ispection type: example:
##teamcity[inspectionType id='indent' name='' description='' category='eslint']
and then error:
##teamcity[inspection typeId='' message='line 10, col 1, Expected indentation of 0 spaces but found 1. (indent)' file='/temp/anyPath/file.js' line='1' additional attribute='ERROR']

**Comment by akaguny** [comment]
@andreogle i'm found another one amazing instrument as projects on github, it's Milestone
Docs for it: https://guides.github.com/features/issues/

**Comment by andreogle** [comment]
thanks. I've used milestones before for v1.3. I think that this will probably be v2.0 since it's a fairly big change from the current behaviour

**Comment by andreogle** [comment]
**Update:**
My Mac was in for repairs and I've been really busy lately, but I've finally managed to get back onto this. The inspections documentation is pretty limited and I haven't found any examples to work from, so it's a lot of trial and error which is time consuming. Made some progress today and this is what it looks like so far:

<img width="856" alt="screen shot 2017-11-28 at 17 40 43" src="https://user-images.githubusercontent.com/4114009/33328776-921e0af6-d463-11e7-8e2d-c1b4c67fd7f8.png">

Hopefully I can get this done sometime soon

**Comment by akaguny** [comment]
it`s cool)
i also warked on another project and have not any time to do feature. Thnks for good work!

**Comment by andreogle** [comment]
Another update:

So this has been a much bigger task than I first thought. I've rewritten most of the project to handle different subformatters (errors and inspections) using ES2015, so now **Node v6.4 or greater is required**.

I've created 2 ways to choose how eslint-teamcity behaves. Nothing is required, but you can set these to customise behaviour. `error count name` and `warning count name` are for the build statistic graph values:

1. Using `package.json` in the root directory:
```
...
"eslint-teamcity": {
  "reporter": "inspections",
  "report-name: "Formatting problems",
  "error-count-name": "My Errors",
  "warning-count-name": "My Warnings",
}
...
```

2. Using ENV variables:
```
export ESLINT_TEAMCITY_REPORTER=inspections
export ESLINT_TEAMCITY_REPORT_NAME="My Formatting Problems"
export ESLINT_TEAMCITY_ERROR_COUNT_NAME="My Error Count"
export ESLINT_TEAMCITY_WARNING_COUNT_NAME="My Warning Count"
```

The defaults (if nothing is specified) are:
```
reporter=errors
reportName="ESLint Violations"
errorCountName="ESLint Error Count"
warningCountName='ESLint Warning Count"
```

I'm going on holiday on Monday and still have breaking tests to fix, so I'm not confident enough to merge into master just yet. **I will fix any problems, the remaining tests and merge when I'm back - early next year.** 

If you would like to test (which would be greatly appreciated), you can specify `"eslint-teamcity": "andreogle/eslint-teamcity#inspections"` in your `package.json`. You might need to include `rm -rf node_modules` in your TeamCity build step as I found that TeamCity was caching old versions. 

**Any feedback would be greatly appreciated!**

**Comment by akaguny** [comment]
Hi, it's greate!
@andreogle, are you save programm interface to push settigs as report-name, error-count-name?

**Comment by andreogle** [comment]
Not sure that I understand correctly, but it's pretty easy to add any extra settings now to either `package.json` or as ENV variables.

The code looks kinda like this for getting settings:
```
// Attempt to load package.json from current directory
const config = JSON.parse(utils.loadPackageJson())['eslint-teamcity'] || {};

const reportType = propNames.reporter || config.reporter || process.env.ESLINT_TEAMCITY_REPORTER || 'errors';
```

**Comment by akaguny** [comment]
@andreogle , i mean that if i require eslint-teamcity im my own module may i send config to this?

**Comment by andreogle** [comment]
ah sorry! yes you can. 

You can pass an object with the following structure when calling the function:
```
{
  reporter: 'inspections', // or 'errors'
  reportName: 'My Formatting Problems,
  errorCountName: 'My Errors',
  warningCountName: 'My Warnings'
}
```

this is the order it looks for config settings:
1. Any config passed to the function, if you use `require('eslint-teamcity')`. This is the second argument
2. `package.json` settings
3. ENV variables
4. Use the default

**Comment by akaguny** [comment]
env variables must have hight priority, because it must be universal instrument to set configuration for all builds in project.

**Comment by akaguny** [comment]
@andreogle, thnks, i shall test this

**Comment by andreogle** [comment]
Great! Thank you

**Comment by akaguny** [comment]
* default options: https://teamcity.jetbrains.com/viewLog.html?buildId=1260127&buildTypeId=TestDrive_Kkk_Build&tab=buildResultsDiv
* change reporter to inspections: 
  * only errors: https://teamcity.jetbrains.com/viewLog.html?buildId=1260131&tab=Inspection&buildTypeId=TestDrive_Kkk_Build&logTab=
  * errors and warnings: https://teamcity.jetbrains.com/viewLog.html?buildId=1260134&buildTypeId=TestDrive_Kkk_Build&tab=Inspection

this teamcity account is trial and links are available near 56 days only

**Comment by andreogle** [comment]
What are these links supposed to be of? I can't login to view them - I don't have any login credentials. Could you maybe show a screenshot rather? What needs to be changed?

**Comment by akaguny** [comment]
all work fine, i don't know how get public access(
![default](https://user-images.githubusercontent.com/3505959/34827875-a4cd4576-f6ec-11e7-9b26-40460217ede1.png)
![default](https://user-images.githubusercontent.com/3505959/34827846-97177974-f6ec-11e7-8221-3f0eae288d90.png)
![default](https://user-images.githubusercontent.com/3505959/34827855-9d3c6e18-f6ec-11e7-9f54-9d3a838bf9b8.png)

if you know how add link to description of rule it will be amazing to implement this feature becouse all of html reports have this

**Comment by andreogle** [comment]
Great! I'll take a look at trying to include links to the rule descriptions

**Comment by andreogle** [comment]
Version 2 has been merged and released!

Unfortunately, I couldn't find any way to get links to the ESLint documentation in the ESLint output. It's also not a reliable way to try to construct the URL using something like `https://eslint.org/docs/rules/<rule-name>`. Seems like TeamCity doesn't have any built in mechanism to include links either. I'm open to suggestions about this, but this is what I could find, sorry!

I made one change to what I mentioned above. When using ENV variables, to set the error/warning names set the following:

not:
```
export ESLINT_TEAMCITY_ERROR_COUNT_NAME="My Error Count"
export ESLINT_TEAMCITY_WARNING_COUNT_NAME="My Warning Count"
```

rather
```
export ESLINT_TEAMCITY_ERROR_STATISTICS_NAME="My Error Count"
export ESLINT_TEAMCITY_WARNING_STATISTICS_NAME="My Warning Count"
```

Thanks again @akaguny 👍
